app/controllers/mixins/ems_common_angular.rb

```diff
@@ -69,7 +73,7 @@ def update_ems_button_validate(verify_ems = nil)
         add_flash(_("Credential validation was not successful: %{details}") % {:details => strip_tags(details)}, :error)
       end
 
-      render :json => {:message => @flash_array.last(1)[0][:message], :level => @flash_array.last(1)[0][:level]}
+      render_flash_json
```

Comment:
So.. we're creating a flash message, only to never actually use it as a flash message, but treat it as a validation status we then render a flash message from, server side.

Seems like this should not be using `add_flash` at all ,and then you don't need to do any @flash_array magic in the first place.

Future refactoring? :)